source/daq/daq_hub/daq_rtc_config.c

```
@@ -0,0 +1,93 @@
+            // Letting the statement fall through is probably fine here?
+        case RTC_CONFIG_STATE_PENDING:
+            if (bGetItemCount(&b_rx_can, RX_TAIL_CAN_RX)) {
+                if ((bGetTailForRead(&b_rx_can, RX_TAIL_CAN_RX, (void**)&buf, &consecutive_items) == 0)) {
+                    if (buf->msg_id == ID_GPS_TIME) {
+                        parse_gps_time(buf, &gps_time);
+                        GPS_time_to_BCD_RTC(&gps_rtc_time, gps_time);
+                        if (!PHAL_configureRTC(&gps_rtc_time, true)) {
+                            // Successful reintialization 
+                            dh.rtc_config_state = RTC_CONFIG_STATE_COMPLETE;
+                        }
+                    }
+                    else {
+                        bCommitRead(&b_rx_can, RX_TAIL_CAN_RX, consecutive_items);
+                    }
+                } else {
+                    dh.can1_rx_overflow++; // currently also incremented by the tcp eth tail people which seems incorrect??
+                }
+            }
+            break;
+        case RTC_CONFIG_STATE_COMPLETE:
+            osThreadTerminate(osThreadGetId());
+            break;        
+    }
+}
+
+static void GPS_time_to_BCD_RTC(RTC_timestamp_t *gps_rtc_time, gps_time_t gps_time)
+{
+    gps_rtc_time->date.year_bcd = RTC_CONV_TO_BCD(gps_time.year);
+    gps_rtc_time->date.month_bcd = RTC_CONV_TO_BCD(gps_time.month);
+    gps_rtc_time->date.day_bcd = RTC_CONV_TO_BCD(gps_time.day);
+
+    gps_rtc_time->time.hours_bcd = RTC_CONV_TO_BCD(gps_time.hour);
+    gps_rtc_time->time.minutes_bcd = RTC_CONV_TO_BCD(gps_time.minute);
+    gps_rtc_time->time.seconds_bcd = RTC_CONV_TO_BCD(gps_time.second);
+
+    gps_rtc_time->date.weekday = RTC_CONV_TO_BCD(get_weekday(gps_time.day, gps_time.month, gps_time.year));
+    gps_rtc_time->time.time_format = RTC_FORMAT_24_HOUR;
+}
+
+static void parse_gps_time(const timestamped_frame_t* frame, gps_time_t* out) {
```

Could be interesting to try to use can_parse.h/.c file to parse? 
Something like casting to the generated `CanParsedData_t` and then doing something like:
```
                CanParsedData_t msg_data = *(CanParsedData_t * )(frame->data);
                out->year   = msg_data->gps_time.year;
                out->month  = msg_data->gps_time.month;
                out->day    = msg_data->gps_time.day;
                out->hour   = msg_data->gps_time.hour;
                out->minute = msg_data->gps_time.minute;
                out->second = msg_data->gps_time.second;
```
(don't know of this exactly will compile)
Not that it should change, but then if TV changes or the CAN message definition changes, your code will continue to still work or fail to compile, instead of fail silently?